File: BarberExpressBackend/src/config/upload.config.ts
// src/config/upload.config.ts
import path from 'path';

export const uploadConfig = {
    directory: path.join(__dirname, '../../uploads'),
    maxSize: 5 * 1024 * 1024, // 5MB
    allowedMimeTypes: ['image/jpeg', 'image/png', 'image/gif'],
    fileFilter: (file: Express.Multer.File): boolean => {
        return uploadConfig.allowedMimeTypes.includes(file.mimetype);
    }
};

File: barberexpressapp/lib/config/upload_config.dart
// lib/config/upload_config.dart
class UploadConfig {
  static const int maxImageSizeBytes = 5 * 1024 * 1024; // 5MB
  static const int maxImageWidth = 1000;
  static const int maxImageHeight = 1000;
  static const int imageQuality = 70;
  
  static const Map<String, String> allowedMimeTypes = {
    'image/jpeg': 'jpg',
    'image/png': 'png',
    'image/gif': 'gif',
  };

  static bool isValidImageSize(int size) {
    return size <= maxImageSizeBytes;
  }

  static bool isValidMimeType(String mimeType) {
    return allowedMimeTypes.containsKey(mimeType);
  }
}